Integrate pivot elevator intake into dev

BrucePetersProgram opened the pull request:
Integrated all of the current alphabot mechanisms with eachother and changed some of their code structure
Currently commands support intaking, zeroing, scoring l2, and scoring l4
We have setpoints for the other scoring positions but we haven't bound them to a button yet

soniashin committed:
Added pivot targets and ratios

BrucePetersProgram committed:
made the pivot zero at 90 degrees with a positive volatge

BrandonBeaver117 committed:
Added eject to intake

nolandatall committed:
Changed a few things on pivot

about to fix the angle instead of position, renamed modes to adapt to pivot

nolandatall committed:
Merge remote-tracking branch 'origin/feat/pivot' into feat/pivot

nolandatall committed:
added placeholders for pivot and superstructure (mainly pivot)

Added some placeholder values so i can make sure that the pivot actually moves. I'll later make sure that the robot can score and then after that I'll add some zeroing code.

nolandatall committed:
Temporarily commented out a few modes in targetState

BrandonBeaver117 committed:
Adding stator limits

nolandatall committed:
Readded stow

readded stow

NoraZitnick committed:
Merge branch 'dev' into feat/pivot

BrandonBeaver117 committed:
merge conflict resolution

NoraZitnick committed:
Merge branch 'dev' into feat/intake

nolandatall committed:
Added temporary button bindings, got rid of the prescore mode in superstructure

Added button bindings (a is to stow, y is to score), and I also got rid of the PRESCORE function because we don't need it

nolandatall committed:
Merge remote-tracking branch 'origin/feat/pivot' into feat/pivot

nolandatall committed:
Finished the (temporary) button bindings for stow and score

I'll change these later

BrandonBeaver117 committed:
Tuning intake voltages and hold triggers

NoraZitnick committed:
Some changes to make it move

NoraZitnick committed:
Tuning and instantiating superstructure object

NoraZitnick committed:
Merge remote-tracking branch 'origin/dev' into feat/pivot

NoraZitnick committed:
Integrate pivot into Bruce's elevator code

BrucePetersProgram committed:
Added zeroing constants

NoraZitnick committed:
change button bindings

NoraZitnick committed:
Merge remote-tracking branch 'origin/feat/pivot' into feat/pivot

BrucePetersProgram committed:
Changed zeroing to work in the generic superstructure

BrandonBeaver117 committed:
Merge remote-tracking branch 'origin/dev' into feat/intake

BrandonBeaver117 committed:
added driver b controls for testing

BrandonBeaver117 committed:
minor formatting

BrucePetersProgram committed:
pivot zeroing works, elevator doesn't

BrucePetersProgram committed:
Merge remote-tracking branch 'origin/feat/pivot' into feat/pivot

BrucePetersProgram committed:
pivot and elevator work

BrucePetersProgram committed:
Merge remote-tracking branch 'origin/dev' into feat/pivot

BrucePetersProgram committed:
attempted setpoints

BrucePetersProgram committed:
Fixed change in constants when merging from dev

BrucePetersProgram committed:
Added some more logging

BrucePetersProgram committed:
Made changes to zeroing thresholds as per pr request

BrucePetersProgram committed:
Changed settings for easier debugging

BrandonBeaver117 committed:
Merge branch 'feat/intake' into donotcommitme

BrucePetersProgram committed:
Changed some constants and made it auto zero

BrandonBeaver117 committed:
funny state management

BrandonBeaver117 committed:
integrated intake command with scoring

added enum capability to determine next superstructure state

added command to set scoring state, and end when done when superstrucutre is in position

intake runs after superstructure is completey in position

BrucePetersProgram committed:
removed some code that conflicted with dev

BrandonBeaver117 committed:
Merge remote-tracking branch 'origin/feat/integrate-pivot-elevator-intake' into feat/integrate-pivot-elevator-intake

BrandonBeaver117 committed:
Merge branch 'feat/pivot' into feat/integrate-pivot-elevator-intake

BrandonBeaver117 committed:
Merge remote-tracking branch 'origin/dev' into feat/integrate-pivot-elevator-intake

BrandonBeaver117 committed:
updated comment

BrandonBeaver117 committed:
Sequencing for intake

BrucePetersProgram committed:
Fixed bugs with code and now it works!!!

there are a few issues. pivot has a really big zeroing issue - not dev ready

BrucePetersProgram committed:
Removed the middle man

I removed superstrucure and went back to a command based approach for controlling the subsystems.

BrucePetersProgram committed:
Fixed bugs

BrucePetersProgram committed:
Merge remote-tracking branch 'origin/dev' into feat/integrate-pivot-elevator-intake

audreypj commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -88,7 +88,7 @@ public void robotPeriodic() {
     CommandScheduler.getInstance().run();
 
-    Threads.setCurrentThreadPriority(false, 10);
+    // Threads.setCurrentThreadPriority(false, 10);
```
Is there a reason this line was commented out?

BrandonBeaver117 replied:
iirc, we were troubleshooting the gripper, and tried to just reduce possible variables that could be causing issues. 

But now, yea, I think now we should add it back.

@BrucePetersProgram comments?

BrucePetersProgram replied:
I think we just commented it out to see if it changed anything. It doesn't change anything and we should add it back. Although I don't entirely understand what it is supposed to do

NoraZitnick committed:
Merge remote-tracking branch 'origin/dev' into feat/integrate-pivot-elevator-intake

NoraZitnick committed:
Change button bindings to work with dev, and are more practical

NoraZitnick committed:
Slower drive

NoraZitnick committed:
Tuning pivot and elevator

NoraZitnick committed:
Merge remote-tracking branch 'origin/dev' into feat/integrate-pivot-elevator-intake

NoraZitnick committed:
Faster and elevator goes down bit more

BrandonBeaver117 committed:
Adding back thread priority

removed earlier due to debugging - but we determined it was likely not the culprit of strange mechanism behavior

BrandonBeaver117 reviewed (approved):
brandon and audrey say "lgtm"
